### DISCUSSION TITLE
How to test information message box using `@vscode/test-electron`?

### DISCUSSION THREAD

**Discussion by PEZ** [microsoft/vscode-discussions#376] [Extension Development QnA] [answered]
In my e2e testing I need to check that message boxes display the right buttons and that these buttons work. Is there a way I can do that?

Update: I'm using this approach to testing: https://code.visualstudio.com/api/working-with-extensions/testing-extension

**Comment by alefragnani** [comment, marked as answer]
Hi @PEZ ,

I'm not sure how you are doing e2e tests, but if you are simply using https://code.visualstudio.com/api/working-with-extensions/testing-extension, I'm also interested on your question. I decided to add tests on some of my extensions a couple of months ago, and to be honest, it was a mining process. 

Almost no extension that I have installed/forked to provide examples, except Peacock, which uses `sinon` and its `stubs` and `mocks` API, but I guess it wouldn't work.

I remember a post in the Slack channel, which suggested https://webdriver.io/docs/wdio-vscode-service/ (posted here too https://github.com/microsoft/vscode-discussions/discussions/9). I also remember another post (I guess on Twitter) about using Cypress and Playwright to test extensions (found this https://medium.com/modern-mainframe/test-automation-with-playwright-for-vs-code-extensions-facilitating-the-growing-interest-in-dcc463f81efa). 

 Maybe that could help

**Reply by PEZ** [reply]
Thanks, @alefragnani! I'm using the approach you linked to first there. Updated both the title and the question text to reflect this.

I would consider trying those other options if this wasn't Joyride I am testing, and there it makes sense to use Joyride itself, which calls for the tests running in VS Code itself. 

In Calva I have tests involving QuickPicks, but there I have a custom QuickPick implementation which is instrumented to facilitate testing. I'm unaware of any similar custom options for message boxes.
